### ISSUE TITLE
need to call margins before compose for child charts to inherit

### ISSUE THREAD

**Issue by rand0m86** [dc-js/dc.js#1003]
I have simple example of using composite chart with currently latest dc js version (18). Anyway if I try to add any margin to chart definition I get chart lines shifted.

``` js
myChart
    .width(500)
    .height(200)
    //.margins({top: 10, right: 50, bottom: 30, left: 40})
    .dimension(timeDimension)
    .compose([
        dc.lineChart(myChart)
            .group(group1, "G1"),
        dc.lineChart(myChart)
            .group(group2, "G2")
            .useRightYAxis(true)
            .ordinalColors(["#ff7f0e"])
    ])
    .x(d3.time.scale().domain([minDate, maxDate]))
    .xUnits(d3.time.weeks)
    .elasticX(true)
    .elasticY(true)
    //.margins({top: 10, right: 50, bottom: 30, left: 40})
    .legend(dc.legend().x(60).y(10).itemHeight(13).gap(5))
    .on("renderlet", rotateTimeAxisLabels);
```

**UPDATE** Depends on position of `.margins(...)` call. If it is at the beginning then everything is ok. If it is at the end of config - I get a bug with shifting.

You can play a little bit with method call position, it depends.

**Comment by gordonwoodhull** [comment]
Thanks @rand0m86, indeed there is some trickiness about when attributes get copied down from the composite chart to its children. The margins only get copied down in `compose` itself. Why some attributes get copied later, during `generateG`, I don't think anyone fully understands.

I'm going to keep this open as a documentation issue - docs should list the current behavior, if not explain it.

**Comment by gordonwoodhull** [comment]
Fixed by #1365 in 3.1.5
